cdap-ui/app/cdap/components/Replicator/Detail/TableScatterPlotGraph/scatterPlot.ts

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as d3 from 'd3';
+import { ITableMetricsData } from 'components/Replicator/Detail/TableScatterPlotGraph/parser';
+import numeral from 'numeral';
+
+export const COLOR_MAP = {
+  active: '#0076DC',
+  activeOutline: '#9FC0FB',
+  inactive: '#999999',
+  inactiveOutline: '#686868',
+  axisColor: '#DDDDDD',
+  legend: '#999999',
+  tick: '#333333',
+};
+
+export function renderScatterPlot(id: string, data: ITableMetricsData[], containerWidth: number) {
+  if (containerWidth < 0) {
+    return;
+  }
+  const containerHeight = 350;
+
+  const margin = {
+    top: 10,
+    bottom: 50,
+    left: 60,
+    right: 35,
+  };
+
+  const width = containerWidth - margin.left - margin.right;
+  const height = containerHeight - margin.top - margin.bottom;
+
+  const svg = d3
+    .select(`#${id} > svg`)
+    .attr('height', containerHeight)
+    .attr('width', containerWidth);
+
+  // Clear out existing graph
+  const groupId = 'replication-latency-graph-con13tainer';
+  const groupElem = d3.select(`#${id} > svg > #${groupId}`);
+  groupElem.remove();
+
+  // Start graph render
+  const chart = svg
+    .append('g')
+    .attr('id', groupId)
+    .attr('transform', `translate(${margin.left}, ${margin.top})`);
+
+  const GRAPH_BUFFER = 1.25;
+  const MIN_VALUE = 1;
+
+  const xMax = Math.max(MIN_VALUE, d3.max(data.map((d) => d.eventsPerMin)));
```

Comment:
Small optimization: `d3.max` takes an accessor. So that portion could be `d3.max(data, (d) => d.eventsPerMin)`